src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -239,17 +241,21 @@ def register_source_uri_arguments(scope):
 for item in ['blob', 'file', 'queue', 'table']:
     register_cli_argument('storage account show-connection-string', '{}_endpoint'.format(item), help='Custom endpoint for {}s.'.format(item))
 
+register_cli_argument('storage account create', 'account_type', help='The storage account type', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='AccountType')))
+
 register_cli_argument('storage account create', 'account_name', account_name_type, options_list=('--name', '-n'), completer=None)
 
-register_cli_argument('storage account create', 'kind', help='Indicates the type of storage account.', **enum_choice_list(Kind))
+register_cli_argument('storage account create', 'kind', help='Indicates the type of storage account.', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='Kind')))
 register_cli_argument('storage account create', 'tags', tags_type)
 
 for item in ['create', 'update']:
-    register_cli_argument('storage account {}'.format(item), 'sku', help='The storage account SKU.', **enum_choice_list(SkuName))
-    register_cli_argument('storage account {}'.format(item), 'encryption', nargs='+', help='Specifies which service(s) to encrypt.', validator=validate_encryption, **enum_choice_list(list(EncryptionServices._attribute_map.keys())))  # pylint: disable=protected-access
+    register_cli_argument('storage account {}'.format(item), 'sku', help='The storage account SKU.', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='SkuName')))
+    es_model = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='EncryptionServices')
+    if es_model:
+        register_cli_argument('storage account {}'.format(item), 'encryption', nargs='+', help='Specifies which service(s) to encrypt.', validator=validate_encryption, **enum_choice_list(list(es_model._attribute_map.keys())))  # pylint: disable=protected-access
```

Comment:
I'm assuming this is the answer to how you handle things that don't exist in all API versions. This is even more cumbersome and hard to get right. Why can't the logic of getting the versioned model apply in "enum_choice_list"?  If you register a parameter for something that doesn't exist it already ignores it, so the only problem would be the exception in the enum_choice_list method, which could just return an empty list.